cachito/workers/tasks/general.py

```diff
@@ -169,6 +171,36 @@ def save_bundle_archive_checksum(request_id: int) -> None:
     bundle_dir.bundle_archive_checksum.write_text(checksum, encoding="utf-8")
 
 
+def _check_packages_data_on_api(
+    request_id: int, packages_count: int, dependencies_count: int
+) -> bool:
+    request = get_request(request_id)
+    packages = request.get("packages", [])
+    dependencies = request.get("dependencies", [])
+
+    return len(packages) == packages_count and len(dependencies) == dependencies_count
+
+
+def _wait_until_packages_file_is_available(
+    request_id: int, packages_count: int, dependencies_count: int
+) -> None:
+    check_passed = False
+    retries = 0
+
+    config = get_worker_config()
+
+    while not check_passed and retries <= config.check_packages_file_max_retries:
+        if retries > 0:
+            time.sleep(config.check_packages_file_interval)
+
+        log.info("Checking if the packages file was properly loaded. Retried %s times.", retries)
+        check_passed = _check_packages_data_on_api(request_id, packages_count, dependencies_count)
+        retries += 1
+
+    if not check_passed:
+        raise CachitoError("Packages file could not be loaded for request %i.", request_id)
```

Comment:
I don't think this will format the string properly, this would for a log message but not for an exception

Reply:
/me facepalms heavily.